tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -185,12 +209,14 @@ def GenerateExamplesByBeam(
       for split in output_config.split_config.splits:
         total_buckets += split.hash_buckets
         buckets.append(total_buckets)
+      # If it's not empty, use a feature for partitioning the examples.
+      feature_name = output_config.split_config.partition_feature_name
```

Comment:
if it's not set, is that None or Empty str? 

I feel it might be better to pass the entire output_config.split_config to _PartitionFn as you will need other fields in the future.

And you can also use proto [HasField](https://developers.google.com/protocol-buffers/docs/reference/python-generated#singular-fields-proto2) to check if it's set or not (in _PartitionFn) instead of using None or Empty

Reply:
When I don't set partition_feature_name, it is an empty string. I used print statements to check this. I can definitely change it to pass the entire output_config.split_config if we need other fields.

Reply:
let's change _PartitionFn to take split_config and use hasField to check existence,
hasField is better than depending on None or Empty (though in python both are false)